akka-http-core/src/main/scala/akka/http/impl/engine/server/HttpServerBluePrint.scala

```diff
@@ -391,11 +391,15 @@ private[http] object HttpServerBluePrint {
               push(requestPrepOut, rs)
             case MessageEnd ⇒
               messageEndPending = false
-              push(requestPrepOut, MessageEnd)
+              if (!oneHundredContinueResponsePending) push(requestPrepOut, MessageEnd)
             case MessageStartError(status, info) ⇒ finishWithIllegalRequestError(status, info)
             case x: EntityStreamError if messageEndPending && openRequests.isEmpty ⇒
               // client terminated the connection after receiving an early response to 100-continue
               completeStage()
+            case x if oneHundredContinueResponsePending ⇒
+              log.debug("Dropping incoming entity since `oneHundredContinueResponsePending` still active")
+              pull(requestParsingIn)
```

Comment:
Will this work? What kind of elements do we discard here? We might end up in an illegal state so it might make sense to just kill the connection?